packagedb/api.py

```diff
@@ -419,6 +420,95 @@ def filter_by_checksums(self, request, *args, **kwargs):
         return self.get_paginated_response(serialized_package_data)
 
 
+class PackageViewSet(PackagePublicViewSet):
+    @action(detail=True)
+    def reindex_package(self, request, *args, **kwargs):
+
+        """
+        Reindex this package instance
+        """
+        package = self.get_object()
+        package.reindex()
+        data = {
+            'status': f'{package.package_url} has been queued for reindexing'
+        }
+        return Response(data)
+
+class PackageUpdateSet(viewsets.ViewSet):
+
+    """
+    Take a list of `packages` (where each item is a dictionary containing PURL
+    and content_type).
+
+    If `uuid` is given then all purls will be added to package set if it exists
+    else a new set would be created and all the purls will be added to that new set.
+
+    **Note:** There is also a slight addition to the logic where a purl already exists in the database
+    and so there are no changes done to the purl entry it is passed as it is.
+
+    **Request example:**
+        {
+          "purls": [
+            {"purl": "pkg:npm/less@1.0.32", "content_type": 1}
+          ],
+          "uuid" : "b67ceb49-1538-481f-a572-431062f382gg"
+        }
+    """
+
+    def create (self, request):
+
+        res = []
+
+        serializer = UpdatePackagesSerializer(data=request.data)
+
+        if not serializer.is_valid():
+            return Response({'errors': serializer.errors}, status=400)
+
+        validated_data = serializer.validated_data
+        packages = validated_data.get('purls', [])
+        uuid = validated_data.get('uuid', None)
+
+        flag = True
+
+        if uuid:
+            pack = PackageSet.objects.filter(uuid=uuid)
+
+            if pack:
+                package_set = pack
+                flag = False
+
+        else:
+            # Create Package Set and set package_set as the same object
+            package_set = PackageSet.objects.create()
+
+        for items in packages or []:
+            temp = {}
```

Comment:
Rename this to `res_data` or something that tells us what it is used for

Reply:
Update the same in code